distributed/ddp/example.py

```diff
@@ -48,16 +50,35 @@ def demo_basic(local_world_size, local_rank):
 
 
 def spmd_main(local_world_size, local_rank):
-    # These are the parameters used to initialize the process group
-    env_dict = {
-        key: os.environ[key]
-        for key in ("MASTER_ADDR", "MASTER_PORT", "RANK", "WORLD_SIZE")
-    }
-    print(f"[{os.getpid()}] Initializing process group with: {env_dict}")
-    dist.init_process_group(backend="nccl")
+    
+    if sys.platform == "win32":
+        # Distributed package only covers collective communications with Gloo
+        # backend and FileStore on Windows platform. Set init_method parameter
+        # in init_process_group to a local file.
+        if "INIT_METHOD" in os.environ.keys():
+            print(f"init_method is {os.environ['INIT_METHOD']}")
+            url_obj = urlparse(os.environ["INIT_METHOD"])
+            if url_obj.scheme.lower() != "file":
+                raise ValueError("Windows only supports FileStore")
+            else:
+                init_method = os.environ["INIT_METHOD"]
+        else:
+            # It is a example application, For convience, we create a file in temp dir.
+            temp_dir = tempfile.gettempdir()
+            init_method = "file:///" + os.path.join(temp_dir, "ddp_example")
+        dist.init_process_group(backend="gloo", init_method=init_method, rank=local_rank, world_size=local_world_size)
```

Comment:
if using `local_rank` and `local_world_size` here, this example cannot be run on multiple servers. Any reason for not using `RANK` and `WORLD_SIZE` env var as for non-Windows cases?

Reply:
Thank your comments. Updated